workcal/angular/src/app/calendar-page/calendar-page.component.ts

```diff
@@ -0,0 +1,482 @@
+import { Component, OnInit } from '@angular/core';
+import { EventApiService } from '../event-api.service';
+import { EventDto, SchedulerEvent , UserDto, UserResponse } from '../models/event-dto.model';
+import { DxSchedulerModule, DxDraggableModule, DxScrollViewModule, DxColorBoxModule  } from 'devextreme-angular';
+import { Router } from '@angular/router';
+import { catchError } from 'rxjs/operators';
+import { of } from 'rxjs';
+import { UserApiService } from '../user-api.service';
+import { Location } from '@angular/common';
+
+@Component({
+  selector: 'app-calendar-page',
+  templateUrl: './calendar-page.component.html',
+  styleUrls: ['./calendar-page.component.scss']
+})
+
+
+export class CalendarPageComponent implements OnInit {
+  events: EventDto[] = [];
+  schedulerEvents: SchedulerEvent[] = [];
+
+  selectedEvent: EventDto;
+
+  currentDate: Date = new Date();
+  currentView: string = 'day';
```

Comment:
it should have a string literal type so if the value array is the following: 'day, 'month', 'year' then the type should only allow that.
currentView: ViewType = 'day' |'month' | 'year'

export type ViewType =